django/cantusdb_project/main_app/signals.py

```diff
@@ -201,3 +205,64 @@ def update_prefix_field(instance):
         instance.__class__.objects.filter(pk=pk).update(prefix=prefix)
     else:  # feast_code is None, ""
         instance.__class__.objects.filter(pk=pk).update(prefix="")
+
+
+def update_chant_incipit_field(chant: Chant):
+    """Update the incipit field of the specified Chant to be the first
+    several words of the chant's standardized-spelling fulltext
+
+    Args:
+        chant (Chant): The chant from the database whose `incipit` field
+        is to be updated
+    """
+    fulltext: Optional[str] = chant.manuscript_full_text_std_spelling
+    if fulltext:  # many chants in the database have only an incipit -
+        # we should only update the incipit if the chant has a fulltext,
+        # just in case a chant manages to get saved without a fulltext somehow
+        new_incipit: str = generate_incipit(fulltext)
+        Chant.objects.filter(id=chant.id).update(incipit=new_incipit)
```

Comment:
So, if a chant doesn't have full text, how do I edit an incipit after this PR since the incipit field is now read-only in the Admin view and not a field in the Chant Edit form?

Reply:
I just tested on staging - in the admin area, you cannot save a chant without specifying a fulltext. It has never been possible in NewCantus to update a chant's incipit without suppling a fulltext, as far as I know.

Reply:
Cool! I just looked and there are only four chants with empty manuscript_full_text_std_spelling in Cantus DB, so yes, not something that happens.

Reply:
> Cool! I just looked and there are only four chants with empty manuscript_full_text_std_spelling in Cantus DB, so yes, not something that happens.

wait, really? I thought there were quite a few chants without a full text (but with an incipit). But the scenario you propose is basically moot: if you want to update an incipit, it's presumably because you know what the text should be, which means you are in a position to give it a full text (or at least to put something in that field), which is desirable anyway. So requiring a full text to change the incipit seems fine...

Reply:
Whelp! looks like I fell into the character-based field null trap (https://docs.djangoproject.com/en/5.0/ref/models/fields/#django.db.models.Field.null) 

There are many chants with null `manuscript_full_text_std_spelling`, but only four with empty string.

So...my bad....nvm.